### ISSUE TITLE
Criterion benchmarks

### ISSUE THREAD

**Issue by chessai** [andrewthad/rotera#7]
With graphs

**Comment by chessai** [comment]
Not sure what data we should benchmark.

**Comment by chessai** [comment]
@andrewthad

I think a good place to start would be benchmarking push without committing, and push with committing, and push with intermittent committing

What do you think? Any other ideas?

**Comment by andrewthad** [comment]
The only way I can imagine doing benchmarks with criterion would be to make the mmap backing the database `MAP_ANONYMOUS` and then to just insert a ton a data. Committing would be a no-op so the frequency of commits would be irrelevant.

A real benchmark, where data is actually persistented, would be trickier to do. I don't think that criterion, which is intended to be used for microbenchmarks, would be suitable for this.

**Comment by chessai** [comment]
What would be suitable? Rolling something on our own?

**Comment by andrewthad** [comment]
Yes, we could still have it running on a single host, but we would just clock a giant write of around a gigabyte of data.
